src/python_testing/TC_EWATERHTR_2_3.py

```diff
@@ -0,0 +1,287 @@
+
+        self.step("3a")
+        await self.check_whm_attribute("HeatDemand", 0)
+
+        self.step("3b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("3c")
+        await self.check_whm_attribute("TankPercentage", 0)
+
+        self.step("3d")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_equal(
+            heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kImmersionElement1)
+
+        self.step("4")
+        await self.send_boost_command(duration=600, target_percentage=100)
+
+        self.step("4a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("4b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("5")
+        await self.send_test_event_trigger_water_temperature61C_test_event()
+
+        self.step("5a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_equal(heatDemand, 0)
+
+        self.step("5b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("5c")
+        await self.check_whm_attribute("TankPercentage", 100)
+
+        self.step("6")
+        await self.send_test_event_trigger_draw_off_hot_water_test_event()
+
+        self.step("6a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
+
+        self.step("6b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
+
+        self.step("6c")
+        await self.check_whm_attribute("TankPercentage", 75)
+
+        self.step("7")
+        await self.send_cancel_boost_command()
+
+        self.step("7a")
+        await self.check_whm_attribute("HeatDemand", 0)
+
+        self.step("7b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("7c")
+        await self.check_whm_attribute("TankPercentage", 75)
+
+        self.step("8")
+        await self.send_boost_command(duration=600, target_percentage=100, target_reheat=65)
+
+        self.step("8a")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_not_equal(
+            heatDemand & Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kImmersionElement1, 0)
```

Comment:
{valhas} greater than 0x00 (demand on at least one source) and (HeaterDemand & (!HeaterTypes)) is zero (demand is only from declared supported types)

Reply:
Done